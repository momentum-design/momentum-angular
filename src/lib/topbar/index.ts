export * from './topbar.module';
export * from './topbar.component';
export * from './topbar-brand.component';
export * from './topbar-mobile.component';
export * from './topbar-nav.component';
export * from './topbar-right.component';
