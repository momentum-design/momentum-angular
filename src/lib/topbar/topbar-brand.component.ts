/** @component topbar-brand */

import { Component, Input } from '@angular/core';

@Component({
  selector: '[md-top-bar-brand]',
  exportAs: 'mdTopbarBrand',
  template: `
    <div class="md-brand__logo">
      <img *ngIf="image" src="{{ image }}" alt="{{ alt }}" />
      <i *ngIf="!image" [ngClass]="['icon', icon]"></i>
    </div>
    <div class="md-brand__title">{{ title }}</div>
  `,
  host: {
    class: 'md-brand',
  },
})
export class TopbarBrandComponent {
  /** @option Image alt text | null */
  @Input() public alt: string;
  /** @option Icon class name | 'icon-cisco-logo' */
  @Input() public icon: string = 'icon-cisco-logo';
  /** @option Image source URL | null */
  @Input() public image: string;
  /** @option Topbar title text | '' */
  @Input() public title: string;
}
