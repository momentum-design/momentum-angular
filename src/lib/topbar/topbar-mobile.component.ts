/** @component topbar-mobile */

import { Component, HostListener, Input } from '@angular/core';

@Component({
  selector: 'md-top-bar-mobile',
  template: `
    <md-icon
      *ngIf="!isMobileOpen"
      name="list-menu_20"
      buttonClassName="md-top-bar__mobile-menu-button"
      [attr.aria-pressed]="isMobileOpen"
      [ariaLabel]="openMenuAriaLabel"
      (click)="handleOpen()"
    ></md-icon>
    <div
      [ngClass]="[
        'md-top-bar__mobile',
        'md-tb-mobile',
        (isMobileOpen && 'open') || ''
      ]"
      (click)="shouldCloseOnClick ? handleClose() : null"
      (keydown)="handleKeydown($event)"
      role="menu"
      tabIndex="0"
    >
      <md-icon
        name="cancel_20"
        buttonClassName="md-tb-mobile__close"
        [attr.aria-pressed]="isMobileOpen"
        [ariaLabel]="closeMenuAriaLabel"
        (click)="handleClose()"
      ></md-icon>
      <span (click)="handleClose()" (keydown)="handleKeydown($event)">
        <div class="md-top-bar__brand">
          <ng-content select="brand"></ng-content>
        </div>
      </span>
      <md-list-separator backgroundColor="transparent"></md-list-separator>
      <nav
        class="md-tb-mobile__nav"
        (click)="!shouldCloseOnClick ? handleClose() : null"
      >
        <ng-content></ng-content>
      </nav>
    </div>
    <div
      [ngClass]="['md-tb-mobile__mask', (isMobileOpen && 'open') || '']"
      (click)="handleClose()"
      role="none"
    ></div>
  `,
})
export class TopbarMobileComponent {
  /** @option Aria Label for close Button | 'Close Menu' */
  @Input() public closeMenuAriaLabel: string = 'Close Menu';
  /** @option Aria Label for open Button | 'Open Menu */
  @Input() public openMenuAriaLabel: string = 'Open Menu';
  /** @option Set mobile menu to close on any click | true */
  @Input() public shouldCloseOnClick: boolean = true;

  public isMobileOpen: boolean = false;

  @HostListener('document:keydown', ['$event']) handleKeydown(
    e: KeyboardEvent
  ) {
    if (e.code === 'Enter') {
      this.handleClose();
      e.preventDefault();
    }
  }

  handleClose(): void {
    this.isMobileOpen = false;
  }

  handleOpen(): void {
    this.isMobileOpen = true;
  }
}
