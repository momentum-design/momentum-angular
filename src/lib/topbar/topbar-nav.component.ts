import { Component } from '@angular/core';

@Component({
  selector: 'md-top-bar-nav',
  template: `
    <md-list tabType="horizontal">
      <ng-content></ng-content>
    </md-list>
  `,
  host: {
    class: 'md-top-bar__nav',
  },
})
export class TopbarNavComponent {}
