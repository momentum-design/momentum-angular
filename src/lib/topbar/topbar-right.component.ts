import { Component } from '@angular/core';

@Component({
  selector: 'md-top-bar-right',
  template: `
    <ng-content></ng-content>
  `,
  host: {
    class: 'md-top-bar__right',
  },
})
export class TopbarRightComponent {}
