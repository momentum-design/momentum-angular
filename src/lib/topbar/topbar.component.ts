import { Component, Input } from '@angular/core';

export type TopbarColor = 'dark' | 'light' | 'blue';

@Component({
  selector: 'header[cui-top-bar]',
  template: `
    <div class="cui-top-bar__container">
      <div class="cui-top-bar__brand">
        <ng-content select="brand"></ng-content>
      </div>
      <ng-content></ng-content>
    </div>
  `,
  host: {
    'attr.role': 'navigation',
    class: 'cui-top-bar',
    '[class.cui-top-bar--fixed]': 'fixed',
    '[class.cui-top-bar--blue]': 'color === "blue"',
    '[class.cui-top-bar--dark]': 'color === "dark"',
    '[class.cui-top-bar--light]': 'color === "light"',
  },
})
export class TopbarComponent {
  /** @prop Topbar header color | '' */
  @Input() public color: TopbarColor;
  /** @prop Determines if Topbar is fixed to top | false */
  @Input() public fixed: boolean;
}
