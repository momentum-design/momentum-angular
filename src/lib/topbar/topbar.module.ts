import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IconModule } from '../icon/icon.module';
import { ListModule } from '../list/list.module';
import { ListItemModule } from '../list-item/list-item.module';
import { ListSeparatorModule } from '../list-separator/list-separator.module';
import { TopbarComponent } from './topbar.component';
import { TopbarBrandComponent } from './topbar-brand.component';
import { TopbarMobileComponent } from './topbar-mobile.component';
import { TopbarNavComponent } from './topbar-nav.component';
import { TopbarRightComponent } from './topbar-right.component';

@NgModule({
  imports: [
    CommonModule,
    IconModule,
    ListModule,
    ListItemModule,
    ListSeparatorModule,
  ],
  declarations: [
    TopbarComponent,
    TopbarBrandComponent,
    TopbarMobileComponent,
    TopbarNavComponent,
    TopbarRightComponent,
  ],
  exports: [
    TopbarComponent,
    TopbarBrandComponent,
    TopbarMobileComponent,
    TopbarNavComponent,
    TopbarRightComponent,
  ],
})
export class TopbarModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: TopbarModule,
      providers: [],
    };
  }
}
