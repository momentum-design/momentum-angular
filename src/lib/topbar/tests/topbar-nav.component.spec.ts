import { Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { TopbarModule } from '../topbar.module';

describe('TopbarNavComponent', () => {
  @Component({
    selector: 'test-app',
    template: `
      <cui-top-bar-nav>
        <div class="testingforTbN"></div>
      </cui-top-bar-nav>
    `,
  })
  class TestAppComponent { }

  let testComponent: TestAppComponent;
  let fixture: ComponentFixture<TestAppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [TopbarModule],
      declarations: [TestAppComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestAppComponent);
    testComponent = fixture.componentInstance;
  });

  it('should create and match snapshot', () => {
    fixture.detectChanges();
    expect(testComponent).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render one TopbarNav', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('cui-top-bar-nav');
    expect(element.className).toContain('cui-top-bar__nav');
  });

  it('should render children', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.testingforTbN');
    expect(element).not.toBeNull();
  });
});
