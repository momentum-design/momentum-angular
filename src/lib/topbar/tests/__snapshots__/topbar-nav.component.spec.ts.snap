// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopbarNavComponent should create and match snapshot 1`] = `
<test-app>
  <md-top-bar-nav
    class="md-top-bar__nav"
  >
    <md-list
      class="md-list md-list--horizontal"
      id="md-list-1"
      ng-reflect-tab-type="horizontal"
      role="list"
      tabtype="horizontal"
    >
      <div
        class="testingforTbN"
      />
    </md-list>
  </md-top-bar-nav>
</test-app>
`;
